### PR TITLE
Add the option to match formal charges to FMCS

### PR DESCRIPTION
**Pull request by greglandrum**
This is a C++-based alternative to #1157 
Also provides some additional control over the MCS parameters from Python by exposing the MCSParameters objects to the Python wrapper.

### CODE REVIEW

File: Code/GraphMol/FMCS/Wrap/rdFMCS.cpp [hunk 1]
```diff
@@ -59,6 +41,31 @@ MCSResult *FindMCSWrapper(python::object mols, bool maximizeBonds,
       p.BondTyper = MCSBondCompareOrderExact;
       break;
   }
+}
+
+MCSResult *FindMCSWrapper(python::object mols, bool maximizeBonds,
```
Comment (bp-kelley) [inline Code/GraphMol/FMCS/Wrap/rdFMCS.cpp]: missing match formal charge here?

Comment (greglandrum) [inline Code/GraphMol/FMCS/Wrap/rdFMCS.cpp]: Na, you can provide it with the parameter structure. I've decided that functions with 135 arguments are a bad idea. :-)

**Comment by bp-kelley** [comment]
Just checking!

----
Brian Kelley

**Comment by bp-kelley** [comment]
Should we add a deprecated message to the old api?

**Comment by greglandrum** [comment]
I don't think it's deprecated per se. At least not yet. I'm just not planning on adding new functionality to it.

**Comment by jbDelafosse** [comment]
Thanks for the hard work @greglandrum

I apologize for never doing this myself in the C++ version

**Comment by greglandrum** [comment]
no worries. making C++ changes is what I do. :-)

**Comment by greglandrum** [comment]
That's an excellent idea. I will do that for the next release.

**Pull request merged**